### PR TITLE
fix: opensearch exporter - update ISM policy for existing indices when retention config changes on rerun

### LINKED ISSUE #17186: Opensearch Retention policy is not applied to existing zeebe indices when enabled
clones #16720

### PR DESCRIPTION
**Pull request by mustafadagher**
## Description

Update lifecycle policy for existing indices when retention config changes on opensearch exporter re-run.

The changes adds `ISM policy` to existing index settings when `retention` is enabled, and removes it when retention is disabled

## Related issues

closes #17186

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java [hunk 1]
```diff
@@ -248,12 +264,174 @@ void shouldUpdateIndexStateManagementPolicy() {
...
     assertThat(updatedMinimumAge).isEqualTo(config.retention.getMinimumAge());
   }
+
+  private boolean export(final Record<?> record) {
+    exporter.export(record);
+    return true;
+  }
+
+  /**
+   * policy change is an asynchronous background process in opensearch, that's why I use awaits
```
Comment (megglos) [inline zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java:282]:
```suggestion
   * policy change is an asynchronous background process in opensearch, that's why we use awaits
```

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestClient.java [hunk 2]
```diff
@@ -165,6 +167,36 @@ private PutIndexStateManagementPolicyRequest createPutIndexManagementPolicyReque
...
   }
 
+  public Optional<IndexISMPolicyDto> explainIndex(final String index) {
+    try {
+      final var request = new Request("GET", "_plugins/_ism/explain/" + index);
+      final var response = restClient.performRequest(request);
+      final TypeReference<Map<String, Object>> mapTypeReference = new TypeReference<>() {};
+      final Map<String, Object> output =
+          MAPPER.readValue(response.getEntity().getContent(), mapTypeReference);
+      final var policy =
```
Comment (megglos) [inline zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestClient.java:177]: nice trick :)

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java [hunk 3]
```diff
@@ -248,12 +264,174 @@ void shouldUpdateIndexStateManagementPolicy() {
...
+      final var record2 = factory.generateRecord(ValueType.JOB);
+
+      await("New record is exported, and existing indices are updated")
+          .atMost(Duration.ofSeconds(30))
+          .until(() -> export(record2));
+
+      // then
+      final var index2 = indexRouter.indexFor(record2);
+      await()
+          .untilAsserted(
```
Comment (megglos) [inline zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java:403]: ❓ no timeout specified so a default one is applied? Do you know how long it waits?

**Review by megglos — APPROVED** [review]
Great progress on getting this solved as well!

We also need to backport this all the way back to 8.2 when opensearch support was added 🙈

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestClient.java (hunk 2 above)
Comment (mustafadagher) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestClient.java:177]: Thanks! I wished I could have done it in a cleaner way though 🙏

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java (hunk 3 above)
Comment (mustafadagher) [reply zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java:403]: the default timeout is `10s`

**Pull request merged**